Web/Edubase.Services/Nomenclature/NomenclatureService.cs

```diff
@@ -27,12 +28,25 @@ public string GetGovernorRoleName(eLookupGovernorRole role, eTextCase textCase =
             return name;
         }
 
-        public string GetEstablishmentsPluralName(GT groupType, eTextCase textCase = eTextCase.TitleCase)
+        public static string GetEstablishmentsPluralName(GT groupType, eTextCase textCase = eTextCase.TitleCase)
         {
-            if (groupType.OneOfThese(GT.ChildrensCentresCollaboration, GT.ChildrensCentresGroup)) return "children's centres".ToTextCase(textCase);
-            else if (groupType.OneOfThese(GT.Federation, GT.Trust)) return "schools".ToTextCase(textCase);
-            else if (groupType.OneOfThese(GT.MultiacademyTrust, GT.SingleacademyTrust, GT.SchoolSponsor)) return "academies".ToTextCase(textCase);
-            else throw new NotImplementedException($"Group type '{groupType}' is not supported for this operation");
+            switch(groupType)
+            {
+                case GT.ChildrensCentresCollaboration:
+                case GT.ChildrensCentresGroup:
+                    return "children's centres".ToTextCase(textCase);
+                case GT.Federation:
+                case GT.Trust:
+                    return "schools".ToTextCase(textCase);
+                case GT.MultiacademyTrust:
+                case GT.SingleacademyTrust:
+                case GT.SchoolSponsor:
+                    return "academies".ToTextCase(textCase);
+                case eLookupGroupType.UmbrellaTrust:
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(groupType),
+                        $"Group type '{groupType}' is not supported for this operation");
```

Comment:
Using `ArgumentOutOfRangeException` for a non-numeric value jumped out to me as unusual, but can't quite place why. 

Perhaps `NotSupportedException` might match the description more closely?

Reply:
Feel free to disregard this observation!

Reply:
Happy to have discussion on this, as it may help broaden understanding:
I use ArgumentOutOfRangeException for something like this (taken from https://www.pluralsight.com/guides/commonly-encountered-exceptions-ArgumentOutOfRangeException)

> The ArgumentOutOfRangeException exception is thrown when the argument passed to a method is not null and contains a value that is not within the set of expected values. This exception type has the properties [ParamName](https://docs.microsoft.com/en-us/dotnet/api/system.argumentexception.paramname?view=netframework-4.7.2) and [ActualValue](https://docs.microsoft.com/en-us/dotnet/api/system.argumentoutofrangeexception.actualvalue?view=netframework-4.7.2) which helps understand the cause of the exception. The ParamName property identifies the parameter name of the invalid argument and the ActualValue property identifies the invalid value if a value is present.

I have added a bit onto the use of the exception to provide the value that is causing issues.

When the issue relates to a parameter passed to a method, I do prefer to use argument exceptions too, as they are written for raising iussues with passed parameters.

Reply:
I think it comes down to it "feeling" strange to consider an enum as having a "range". 

If implementing fresh I think I would have defaulted to using either `NotSupportedException` re: the implementation not yet existing, or `ArgumentException`/`InvalidEnumArgumentException` if it's not an appropriate value passed in.

Not that it particularly matters as in this case it's all a bit academic beyond flagging there's an error here 😀 

_(I don't have rights to mark this comment as resolved - feel free to do it on my behalf here!)_

Reply:
Changed to InvalidEnumArgumentException